src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/TopConsumerSummary.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.FlowUnitMessage;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.TopConsumerSummaryMessage;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.GenericSummary;
+import java.util.ArrayList;
+import java.util.List;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+
+/**
+ * TopConsumerSummary contains the name and usage of a resource consumer.
+ */
+public class TopConsumerSummary extends GenericSummary {
+
+  private final String name;
+  private final double value;
+
+  public TopConsumerSummary(final String name, final double value) {
+    super();
+    this.name = name;
+    this.value = value;
+  }
+
+  public String getName() {
+    return this.name;
+  }
+
+  public double getValue() {
+    return this.value;
+  }
+
+  @Override
+  public TopConsumerSummaryMessage buildSummaryMessage() {
+    final TopConsumerSummaryMessage.Builder summaryMessageBuilder = TopConsumerSummaryMessage.newBuilder();
+    summaryMessageBuilder.setName(this.name);
+    summaryMessageBuilder.setValue(this.value);
+    return summaryMessageBuilder.build();
+  }
+
+  @Override
+  public void buildSummaryMessageAndAddToFlowUnit(FlowUnitMessage.Builder messageBuilder) {
```

Comment:
We should probably add a comment why this method is empty

Reply:
done